bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+
+import numpy as np
+import h5py
+import operator
+import argparse
+import logging
+from awkde import GaussianKDE
+from pycbc import init_logging
+import pycbc.conversions as convert
+from sklearn.model_selection import KFold
+
+parser = argparse.ArgumentParser(description=__doc__)
+parser.add_argument('--signal-file')
+parser.add_argument('--template-file', required=True)
+parser.add_argument('--output-file', required=True)
+parser.add_argument('--make-signal-kde', action='store_true')
+parser.add_argument('--make-template-kde', action='store_true')
+parser.add_argument('--verbose', action='count')
+args = parser.parse_args()
+init_logging(args.verbose)
+
+def kde_awkde(x, x_grid, alp=0.5, gl_bandwidth='silverman', ret_kde=False):
+    kde = GaussianKDE(glob_bw=gl_bandwidth, alpha=alp, diag_cov=True)
+    kde.fit(x)
+    if isinstance(x_grid, (list, tuple, np.ndarray)) == False:
+        y = kde.predict(x_grid)
+    else:
+        y = kde.predict(x_grid)
+    if ret_kde == True:
+        return kde, y
+    return y
+
+def kfcv_awkde(sample, bwchoice, alphachoice, k=2):
+    if bwchoice not in ['silverman', 'scott']:
+        bwchoice = float(bwchoice)
+    fomlist =[]
+    kfold = KFold(k, True, None)
+    for train_index, test_index in kfold.split(sample):
+        train, test = sample[train_index], sample[test_index]
+        y = kde_awkde(train, test, alp=alphachoice, gl_bandwidth=bwchoice)
+        fomlist.append(np.sum(np.log(y)))
+    return np.mean(fomlist)
+
+def optimizedparam(sampleval):
+    bwgrid = ['scott', 'silverman']+np.logspace(-2,0,10).tolist()
+    alphagrid = [1]
+    FOM= {}
+    for gbw in bwgrid:
+        for alphavals in alphagrid:
+            FOM[(gbw, alphavals)] = kfcv_awkde(sampleval, gbw, alphavals)
```

Comment:
Here, because of the default choice k=2 in kfcv_awkde it is using 2-fold cross validation for everything (both the signal and template KDEs). 
I think it would be good to have the possibility to use other choices.  So can you add k as a kwarg to this function, and pass the value of k to kfcv_awkde ?
Then eventually we could use different values of k depending on the number of sample points (considering what is the computational load required).